tests/samples/matmul_fill_static_i32_config.mlir

```diff
@@ -0,0 +1,39 @@
+// RUN: iree-opt %s --pass-pipeline="builtin.module(hal.executable(hal.executable.variant(iree-hal-translate-target-executable-variants{target=amd-aie})))" | FileCheck %s
+
+// CHECK-LABEL: hal.executable.export public @matmul_static_dispatch_0_matmul_8x8x16_i32
+//       CHECK:    aie.device(ipu)
+//       CHECK:    aie.shim_dma_allocation
+//       CHECK:    aie.shim_dma_allocation
+//       CHECK:    aie.shim_dma_allocation
+//       CHECK:    func.func @matmul_static_dispatch_0_matmul_8x8x16_i32(%arg0: memref<8x16xi32>, %arg1: memref<16x8xi32>, %arg2: memref<8x8xi32>)
+//       CHECK:      aiex.ipu.dma_memcpy_nd
+//       CHECK:      aiex.ipu.dma_memcpy_nd
+//       CHECK:      aiex.ipu.dma_memcpy_nd
+//       CHECK:      aiex.ipu.sync
+#config = #iree_codegen.lowering_config<tile_sizes = [[8, 8], [4, 4], [0, 0, 4]]>
+#translation = #iree_codegen.translation_info<CPUDefault>
```

Comment:
This is just to demonstrate that `TransformDialectCodegen` is not being set and the e2e still works.

Reply:
Just add this as a comment to the test itself.